server/src/test/java/io/crate/integrationtests/DiskThresholdDeciderIT.java

```diff
@@ -0,0 +1,356 @@
+        assertThat(defaultFileSystem).isNull();
+        defaultFileSystem = PathUtils.getDefaultFileSystem();
+        assertThat(fileSystemProvider).isNull();
+        fileSystemProvider = new TestFileSystemProvider(defaultFileSystem, createTempDir());
+        PathUtilsForTesting.installMock(fileSystemProvider.getFileSystem(null));
+    }
+
+    @After
+    public void removeFilesystemProvider() {
+        fileSystemProvider = null;
+        assertThat(defaultFileSystem).isNotNull();
+        PathUtilsForTesting.installMock(defaultFileSystem); // set the default filesystem back
+        defaultFileSystem = null;
+    }
+
+    private static final long WATERMARK_BYTES = new ByteSizeValue(10, ByteSizeUnit.KB).getBytes();
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        final Path dataPath = fileSystemProvider.getRootDir().resolve("node-" + nodeOrdinal);
+        try {
+            Files.createDirectories(dataPath);
+        } catch (IOException e) {
+            throw new AssertionError("unexpected", e);
+        }
+        fileSystemProvider.addTrackedPath(dataPath);
+        return Settings.builder()
+                .put(super.nodeSettings(nodeOrdinal))
+                .put(Environment.PATH_DATA_SETTING.getKey(), dataPath)
+                .put(FsService.ALWAYS_REFRESH_SETTING.getKey(), true)
+                .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_LOW_DISK_WATERMARK_SETTING.getKey(), WATERMARK_BYTES + "b")
+                .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_WATERMARK_SETTING.getKey(), WATERMARK_BYTES + "b")
+                .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_WATERMARK_SETTING.getKey(), "0b")
+                .put(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_REROUTE_INTERVAL_SETTING.getKey(), "0ms")
+                .build();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.singletonList(InternalSettingsPlugin.class);
+    }
+
+    public void testHighWatermarkNotExceeded() throws InterruptedException, ExecutionException {
+        internalCluster().startMasterOnlyNode();
+        internalCluster().startDataOnlyNode();
+        final String dataNodeName = internalCluster().startDataOnlyNode();
+
+        final InternalClusterInfoService clusterInfoService
+                = (InternalClusterInfoService) internalCluster().getMasterNodeInstance(ClusterInfoService.class);
+        internalCluster().getMasterNodeInstance(ClusterService.class).addListener(event -> clusterInfoService.refresh());
+
+        final String dataNode0Id = internalCluster().getInstance(NodeEnvironment.class, dataNodeName).nodeId();
+        final Path dataNode0Path = internalCluster().getInstance(Environment.class, dataNodeName).dataFiles()[0];
+
+        String indexName = "test";
+        execute("create table " + indexName + "(x text) " +
+            "clustered into 6 shards " +
```

Comment:
taking all settings which are [passed directly](https://github.com/elastic/elasticsearch/commit/3389d5ccb25101b6a81523de4a293a65b082b4bc#diff-e44f19e24fcf88d32bd88d5ed8c4f19b7eb5c01560fc20168784476143cef808R144) to the `createIndex` in the original patch.